### PR TITLE
Added modal reactors documentation

### PR DESCRIPTION
**Pull request by a-sr**
Added some documentation for modal models.

The described behavior for reset, startup, shutdown triggers is not yet implemented but I will tackle this next.
Either we postpone this PR or add a note in the documentation.

### CODE REVIEW

**Comment by a-sr** [comment]
I was not able to test the website locally because `yarn install` failed on my machine (Ubuntu) with the following error:
`Error: gatsby-plugin-client-side-redirect@https://github.com/orta/gatsby-plugin-client-side-redirect.git#commit=44504303249ec7210d816a23c06985c7d9f32772: EINVAL: invalid argument, rename '/tmp/xfs-3e591172' -> '/tmp/xfs-3e591172/package.tgz'`
I have no idea what caused this or how to fix it.

File: packages/documentation/copy/en/topics/Modal Models.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+### Startup and Shutdown
+
+**This behavior may be subject to change in the future**
+
+A challenge for modal execution is the handling `startup` and `shutdown` behavior.
+These are commonly used for managing memory for state variables, handling connections to sensors or actuators, or starting/joining external threads.
+If reactions to these triggers are located inside modes they are subject to a special execution regime.
+
+First, `startup` reactions are invoked at most once at the first activation of a mode.
+Second, `shutdown` reactions are executed when the reactor shuts down, ***irrespective*** of mode activity, but we only if the enclosing modes have been activate at least once.
```
Comment (edwardalee) [inline packages/documentation/copy/en/topics/Modal Models.md:158]:
```suggestion
Second, `shutdown` reactions are executed when the reactor shuts down, ***irrespective*** of mode activity, but only if the enclosing modes have been activated at least once.
```

File: packages/documentation/copy/en/topics/Modal Models.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+Hence, every startup has a corresponding shutdown.
+Third, as mentioned before, the new `reset` trigger for reactions can be used, if a startup behavior should be re-executed if a mode is entered with a reset transition.
+
+Note that this may have unexpected consequences:
+
+* Startup behavior inside modes may occur during execution and not only at program start.
+* Multiple shutdown reactions may be executed, bypassing mutual exclusion of modes.
+* Reactors that are design without consideration of modes and use only `startup` (not `reset`) to trigger an execution chain, may not work in modes and cease to function if re-entered with a reset.
```
Comment (edwardalee) [inline packages/documentation/copy/en/topics/Modal Models.md:166]:
```suggestion
* Reactors that are designed without consideration of modes and use only `startup` (not `reset`) to trigger an execution chain, may not work in modes and cease to function if re-entered with a reset.
```

File: packages/documentation/copy/en/topics/Modal Models.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+As a consequence, a mode has a notion of local time that elapses only when the mode is active.
+
+### Local Time
+
+From the perspective of timers and actions, time is suspended when a mode is inactive.
+This also applies to indirectly nested reactors within modes and connections with logical delays, if their source lies within a mode.
+
+<img alt="Illustration of local time (model)" src="../../../../../img/modal_models/local_time.svg" width="400"/>
+
+The given LF model illustrates the different characteristics of local time affecting timers and actions in the presence of the two transition types.
```
Comment (edwardalee) [inline packages/documentation/copy/en/topics/Modal Models.md:118]:
```suggestion
The above LF model illustrates the different characteristics of local time affecting timers and actions in the presence of the two transition types.
```

File: packages/documentation/copy/en/topics/Modal Models.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+After one second, a mode switch is triggered periodically with a period one second.
+Each mode has a timer `T1`/`T2` that triggers a reaction after an initial offset of 100 msec and then periodically after 750 msec.
+This reaction then schedules a logical action with a delay of 500 msec (the actual target code does not add an additional delay over the minimum specified).
+This action triggers the second reaction, which writes to the output `out`.
+The main difference between the modes is that `One` is entered via a history transition, continuing its behavior, while `Two` is reset.
+(Continue behavior is indicated by an "H" on the transition edge because it enters into the entire history of the mode.)
+
+<img alt="Illustration of local time (trace)" src="../../../../../img/modal_models/local_time_trace.svg" width="600"/>
+
+This is the execution trace of the first 4 seconds of this program.
```
Comment (edwardalee) [inline packages/documentation/copy/en/topics/Modal Models.md:131]:
```suggestion
Above is the execution trace of the first 4 seconds of this program.
```

File: packages/documentation/copy/en/topics/Modal Models.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+Below the timeline is the currently active mode and above the timeline are the model elements that are executed at certain points in time, together with indicating triggering and their relation through time.
+For example, at 100 msec, the initial offset of timer `T1` elapses, which leads to the scheduling of the logical action in this mode.
+The action triggers the reaction 500 msec later, at 600 msec, and thus causes an output.
+The timing diagram illustrates the different handling of time between history transitions and reset transitions.
+Specifically, when mode `One` is re-entered via a history transition, at time 2000 msec, the action triggered by `T1` before, at time 850 msec, resumes.
+In contrast, when mode `Two` is re-entered via a reset transition, at time 3000 msec, the action triggered by `T2` before, at time 1850 msec, gets discarded.
+
+<img alt="Illustration of local time (plot)" src="../../../../../img/modal_models/local_time_plot.svg" width="300"/>
+
+This plot illustrates the relation between global time in the environment and the localized time for each timer in the model.
```
Comment (edwardalee) [inline packages/documentation/copy/en/topics/Modal Models.md:141]:
```suggestion
The above plot illustrates the relation between global time in the environment and the localized time for each timer in the model.
```

**Review by edwardalee — APPROVED** [review]
LGTM!  left some minor comments.

**Comment by a-sr** [comment]
Thanks! I applied you suggestions.

**Pull request merged**